lib/core/src/Cardano/Wallet.hs

```diff
@@ -2186,6 +2186,7 @@ updateCosigner
         , Typeable n
         , WalletKey k
         , HasDBLayer IO s k ctx
+        , k ~ SharedKey
```

Comment:
Just wondering aloud, could the constraints in these functions be made shorter because the types are basically fixed?

Reply:
we can think of introducing constriant groups probably but I am not sure and do not know if they will be more clear. Personally, I am accustomed to explicit constrains. But I was one day looking at ledger repo and they introduced those composite constraints and I think they serve well, but they have situation that one the same composite constraints is in many places...